python/pyspark/sql/pandas/types.py

```diff
@@ -174,6 +257,18 @@ def from_arrow_type(at: "pa.DataType", prefer_timestamp_ntz: bool = False) -> Da
         spark_type = DayTimeIntervalType()
     elif types.is_list(at):
         spark_type = ArrayType(from_arrow_type(at.value_type, prefer_timestamp_ntz))
+    elif types.is_fixed_size_list(at):
+        import pyarrow as pa
+
+        if LooseVersion(pa.__version__) < LooseVersion("14.0.0"):
```

Comment:
we'd better support version >= 4.0.0

https://spark.apache.org/docs/latest/api/python/getting_started/install.html#dependencies